### PR TITLE
bpo-25130: Add calls of gc.collect() in tests to support PyPy

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue25130

### CODE REVIEW

**Review by gpshead — CHANGES_REQUESTED** [review]
TL;DR - We should include comments on all of these stating why they exist and how to know when they're still relevant.

Calling gc.collect(), even multiple times, from any code, test or not, is usually a sign of fragility and something that isn't being done right.  It can't guarantee any particular direct action.

When they don't make a difference in CPython if someone removes them, we can expect these to disappear from the codebase over time during maintenance.

If we want these, we should include a comment on all of them stating why it is here and how to determine if each is still relevant or not.  Otherwise they become mysterious lore in the code and get copied around, deleted, changed, for no rhyme of reason over time.

**Comment by serhiy-storchaka** [comment]
What comments should I add? In most cases the pattern is the same:
```
wr = weakref.ref(obj)
del obj
gc.collect()
self.assertIsNone(wr())
```
It is all either to check that there are no other references to obj, or to trigger it's `__del__`.

**Comment by gpshead** [comment]
Even just an end of line comment on the added calls such as

```
gc_collect()  # For pypy or other GCs.
```

So long as there's some indication of why the call is there given that today's CPython refcounting gc didn't need it.

**Comment by serhiy-storchaka** [comment]
I have made the requested changes; please review again.

**Review by gpshead — APPROVED** [review]
